src/utils/starknetId.ts

```diff
@@ -132,7 +132,7 @@ export const StarknetIdContract = {
 /**
  * Returns the Starknet ID contract address based on the provided chain ID.
  *
- * @param {StarknetChainId} chainId The chain ID of the Starknet network.
```

Comment:
Using the `StarknetChainId` in your previous comment as the input parameter type restricts the input value for functions that depend on it. This becomes an issue in [src/provider/extensions/starknetId.ts](https://github.com/starknet-io/starknet.js/src/provider/extensions/starknetId.ts) because the `getChainId` function returns `ChainId` and not `StarknetChainId`.
Should I still go ahead with this change? @penovicp

Reply:
Yes. Cast the `getChainId` calls as `StarknetChainId` in that file.

Reply:
Changing the input type from `ChainId` to `StarknetChainId` for the function in [src/utils/starknetId.ts](https://github.com/starknet-io/starknet.js/src/utils/starknetId.ts) will result in changing some of the interfaces. After that, the code returns to its original state, allowing only the chain ID of mainnet and sepolia testnet. That defeats the purpose of this change
What do you think?

Reply:
I understand better now. Thank you @penovicp